### DISCUSSION TITLE
Switching flux direction in interface kernel

### DISCUSSION THREAD

**Discussion by lanzam213** [idaholab/moose#23598] [Q&A Modules: General] [answered]
I developed an interface kernel that calculates a flux based on the neighbor value and applies that flux to primary and secondary sides as shown below. 

```
ADReal
ADPelletInterfaceKernel::computeQpResidual(Moose::DGResidualType type)
{

  // positive sign indicates positive flux outward from secondary element (located on left side of interface)
  
  ADReal flux = (_neighbor_value[_qp] > _sat ? _k * (_neighbor_value[_qp] - _sat): 0);
  
  switch (type)
  {
    case Moose::Element:
      return -flux * _stoich_factor * _test[_i][_qp];

    case Moose::Neighbor:
      return flux * _test_neighbor[_i][_qp];

    default:
      return 0.0;
  }
}
```

The interface kernel works perfectly when the boundary is on the right face of secondary side but not when switched. The problem I'm having is when the primary and secondary elements switch places on the interface boundary.  I'm looking for a way to evaluate the element normal such that I can flip the sign on flux when needed. Any suggestions will be appreciated.

-Mark

UPDATE: New testing indicates this is working as is.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You should have access to `_current_elem`, `_current_side` and `_current_elem_side` in a DGKernel to make these determinations. 
You could add a parameter to store a normal if it s relatively consistent on the boundary.

Guillaume
